experimental/list_formatter/src/deduplicating_array.rs

```diff
@@ -0,0 +1,253 @@
+use serde::de::{Deserialize, Deserializer, EnumAccess, Visitor};
+use serde::ser::{Serialize, Serializer};
+
+pub fn serialize<S, T, const N: usize>(array: &[T; N], serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+    T: Serialize + PartialEq,
+{
+    use serde::ser::SerializeTuple;
+
+    let human = serializer.is_human_readable();
+
+    let mut seq = serializer.serialize_tuple(N)?;
+
+    for i in 0..N {
+        match array.iter().take(i).position(|j| &array[i] == j) {
```

Comment:
Nit: `j` is an element in the closure, not an index.

```suggestion
        match array.iter().take(i).position(|value| &array[i] == value) {
```